pandas/core/generic.py

```diff
@@ -2047,7 +2047,7 @@ def __setstate__(self, state):
         elif len(state) == 2:
             raise NotImplementedError("Pre-0.12 pickles are no longer supported")
 
-        self._item_cache = {}
+        self._item_cache: Dict[Hashable, Series] = {}
```

Comment:
I think this will prevent `self._item_cache["somestring"]`, could avoid this with:

```py
from typing import MutableMapping
from pandas._typing import HashableT

self._item_cache: dict[HashableT, Series] = {}
# or with
self._item_cache: MutableMapping[Hashable, Series] = {}
```

Reply:
Thanks for your review!

Not sure I follow though, how does it prevent that?
```python
(.venv) megorelli@penguin:~/pandas-dev$ cat t.py 
from typing import Hashable, Dict

foo: Dict[Hashable, int] = {}

foo['somestring']
(.venv) megorelli@penguin:~/pandas-dev$ mypy t.py 
Success: no issues found in 1 source file
```

Reply:
Sorry, confused it with the case when dict is an argument of a function:

```py
from typing import (
    Dict,
    Hashable,
)

def test(foo: Dict[Hashable, int]):
    ...

foo: Dict[str, int] = {}
test(foo)  # error
```